app/src/main/java/com/alphawallet/app/ui/ImportKeystoreFragment.java

```diff
@@ -75,7 +75,10 @@ private void setupView()
     public void onResume()
     {
         super.onResume();
-        if (keystore == null && getActivity() != null) setupView();
+        if (keystore == null || password == null)
+        {
+            requireActivity().recreate();
```

Comment:
Is it possible for this to get into an infinite recursion state?

Reply:
> Is it possible for this to get into an infinite recursion state?

This is a signal to the OS to re-create the underlying activity and re-populate the fragment. It happens if the user has paged out from the application while on this page, done something memory intensive like playing a game, watching youtube, navigation etc and paged back in. While they were doing their other activity the OS scavenged the objects in the class, so now they're back, the objects are all null pointers. After the OS re-instantiates the Activity and Fragment the viewodel will have been initialised again.

Really, the Android OS should provide a formal way to deal with this! If they don't already and I don't know about it.